src/test/java/org/opensearch/security/configuration/ConfigurationRepositoryTest.java

```diff
@@ -111,4 +151,165 @@ public void getConfiguration_withInvalidConfigurationShouldReturnNewEmptyConfigu
+
+        configRepository.initDefaultConfiguration().get();
+
+        verify(securityIndexHandler, atMostOnce()).createIndex(any());
+        verify(securityIndexHandler, atMostOnce()).uploadSecurityConfiguration(any(), any());
+
+        final var clusterStateUpdateTaskCaptor = ArgumentCaptor.forClass(ClusterStateUpdateTask.class);
+
+        verify(clusterService, atMostOnce()).submitStateUpdateTask(anyString(), clusterStateUpdateTaskCaptor.capture());
+
+        assertClusterState(clusterStateUpdateTaskCaptor);
+    }
+
+    @Test
+    public void initDefaultConfiguration_shouldReUploadConfigIfIndexCreated() throws Exception {
+        System.setProperty("security.default_init.dir", Path.of(".").toString());
+        setupClusterService();
+
+        when(clusterService.state().metadata().hasIndex(anyString())).thenReturn(true);
+
+        ConfigurationRepository configRepository = createConfigurationRepository(Settings.EMPTY);
+
+        doAnswer(invocation -> {
+            @SuppressWarnings("unchecked")
+            final var listener = (ActionListener<Set<SecurityConfig>>) invocation.getArgument(1);
+            listener.onResponse(Set.of(new SecurityConfig(CType.CONFIG, "aaa", null)));
+            return null;
+        }).when(securityIndexHandler).uploadSecurityConfiguration(any(), any());
+
+        configRepository.initDefaultConfiguration().get();
+
+        verify(securityIndexHandler, never()).createIndex(any());
+        verify(securityIndexHandler, atMostOnce()).uploadSecurityConfiguration(any(), any());
+
+        final var clusterStateUpdateTaskCaptor = ArgumentCaptor.forClass(ClusterStateUpdateTask.class);
+
+        verify(clusterService, atMostOnce()).submitStateUpdateTask(anyString(), clusterStateUpdateTaskCaptor.capture());
+
+        assertClusterState(clusterStateUpdateTaskCaptor);
+    }
+
+    @Test
+    public void reloadConfiguration_shouldReloadConfigurationOnlyOnce() throws Exception {
+        ConfigurationRepository configRepository = createConfigurationRepository(Settings.EMPTY);
+
+        doAnswer(invocation -> {
+            @SuppressWarnings("unchecked")
+            final var listener = (ActionListener<Map<CType, SecurityDynamicConfiguration<?>>>) invocation.getArgument(1);
+            listener.onResponse(Map.of());
+            return null;
+        }).when(securityIndexHandler).loadConfiguration(any(), any());
+
+        configRepository.reloadConfiguration(new SecurityMetadata(Instant.now(), Set.of(new SecurityConfig(CType.CONFIG, "aaa", null))))
+            .get();
+
+        verify(securityIndexHandler, atMostOnce()).loadConfiguration(any(), any());
+
+        reset(securityIndexHandler);
+
+        assertNull(
+            configRepository.reloadConfiguration(new SecurityMetadata(Instant.now(), Set.of(new SecurityConfig(CType.CONFIG, "aaa", null))))
+        );
+
+        verify(securityIndexHandler, never()).loadConfiguration(any(), any());
```

Comment:
Addressed

Reply:
`verifyNoMoreInteractions(...)` still seems to be part of each individual test. Maybe the code-commit push was missed?

Reply:
I added verifyNoMoreInteractions for all tests except: 

- testClusterChanged_shouldInitSecurityIndexIfNoSecurityData
- testClusterChanged_shouldExecuteInitialization
- testClusterChanged_shouldNotExecuteInitialization

since all of them do `doCallRealMethod`